src/modules/dashboard/components/Whitelist/Whitelist.tsx

```diff
@@ -41,32 +22,10 @@ const Whitelist = ({ colony: { colonyAddress }, colony }: Props) => {
   const { data: usersData, loading: usersLoading } = useWhitelistedUsersQuery({
     variables: { colonyAddress },
   });
-  const { walletAddress, username, ethereal } = useLoggedInUser();
-  const hasRegisteredProfile = !!username && !ethereal;
-  const allUserRoles = useTransformer(getAllUserRoles, [colony, walletAddress]);
-  const userHasPermission = hasRegisteredProfile && hasRoot(allUserRoles);
-
-  const openAgreementDialog = useDialog(AgreementDialog);
-  const {
-    data: agreementHashData,
-    loading: agreementHashLoading,
-  } = useWhitelistAgreementHashQuery({
-    variables: { colonyAddress },
-    fetchPolicy: 'network-only',
-  });
-
-  const openDialog = useCallback(
-    () =>
-      agreementHashData?.whitelistAgreementHash &&
-      openAgreementDialog({
-        agreementHash: agreementHashData?.whitelistAgreementHash,
-      }),
-    [openAgreementDialog, agreementHashData],
-  );
 
   return (
     <div>
-      <UploadAddressesWidget />
+      <UploadAddressesWidget colony={colony} />
       {usersLoading && <MiniSpinnerLoader loadingText={MSG.loadingText} />}
```

Comment:
So my only note here is that I would like to see a loading spinner here, after the last address upload transaction succeeds, so that I know the list needs to be updated.

However, that is actually hard to implement in the current state of the dapp. We'll have to wait and see how long it takes _(in production and on QA)_ from the time you add addresses to the list, to the time it takes for them to show up.

If the gap is not that big, we might not want to bother.